src/TypeScript/DefinitionWriter.cpp

```diff
@@ -654,7 +698,7 @@ std::string DefinitionWriter::tsifyType(const Type& type)
             return "string";
         }
         else if (interface.name == "NSDate") {
```

Comment:
We pass NSDate-s everywhere in the tns-core-modules. I too think we should preserve Date at some places but it can't be that naive as it is at the moment. We should distinct marshaling for in, out and in/out boundaries, and probably out NSDate types should be marshaled as Date everywhere except for factory methods where NSDate should be returned (similar to the stringWithString returning NSString instead of JavaScript string). In parameters should be of type `Date | NSDate` because Date is automatically converted to NSDate so both are accepted there. And for in/out places (such as property types) we should probably use something like Date & NSDate or Date | any or whatever would allow you to get the value and use it as Date but also allow you to set the property using Date or NSDate.